recipes/glslang/all/patches/conanize-cmake-11.5.0.patch

```diff
@@ -0,0 +1,41 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -306,10 +306,6 @@
+     add_subdirectory(External)
+ endif()
+ 
+-if(NOT TARGET SPIRV-Tools-opt)
+-    set(ENABLE_OPT OFF)
+-endif()
+-
+ if(ENABLE_OPT)
+     message(STATUS "optimizer enabled")
+     add_definitions(-DENABLE_OPT=1)
+--- a/SPIRV/CMakeLists.txt
++++ b/SPIRV/CMakeLists.txt
+@@ -91,14 +91,7 @@
+ endif()
+ 
+ if(ENABLE_OPT)
+-    target_include_directories(SPIRV
+-        PRIVATE ${spirv-tools_SOURCE_DIR}/include
+-        PRIVATE ${spirv-tools_SOURCE_DIR}/source
+-    )
+-    target_link_libraries(SPIRV PRIVATE MachineIndependent SPIRV-Tools-opt)
+-    target_include_directories(SPIRV PUBLIC
+-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../External>
+-        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/External>)
++    target_link_libraries(SPIRV glslang ${CONAN_LIBS})
+ else()
+     target_link_libraries(SPIRV PRIVATE MachineIndependent)
+ endif(ENABLE_OPT)
+--- a/glslang/CMakeLists.txt
++++ b/glslang/CMakeLists.txt
+@@ -167,7 +167,6 @@ set(GLSLANG_HEADERS
+ add_library(glslang ${LIB_TYPE} ${BISON_GLSLParser_OUTPUT_SOURCE} ${GLSLANG_SOURCES} ${GLSLANG_HEADERS})
+ set_target_properties(glslang PROPERTIES
+     FOLDER glslang
+-    POSITION_INDEPENDENT_CODE ON
```

Comment:
Is this not covered by...

Reply:
Yes you are right, but removing the target_link_libraries command ensues in linking errors for glslangValidator.exe which depends on glslang, and that's because OSDependent, OGLCompiler and MachineIndependent targets aren't linked in as glslangValidator is built from its own CMakeLists.txt in /StandAlone/. I might make a change to the patch to add this target to glslangValidator in StandAlone/CMakeLists.txt so it compiles (I tested it), but what's the best way in this case?

Reply:
🤯 I think that part of the patch is fine... my comment was related to the `fPIC` handling which is getting deleted twice in this file 🙊 sorry for the confusion... I do not know this library well enough to comment on it's linkage requirements

Reply:
Oh! My bad, I didn't say the little arrow above the comment clearly pointing at the fPIC stuff. In 11.5.0, only for the glslang target, the set_property is a multiline command changing a couple more properties for the target, and thus replace_in_file would fail. I thought of having a different replace_in_file only for that target, but it was easier adding the change to the patch file (no space/line feed issue). I did protect the glslang with an if on the version indeed.

Reply:
Thats a good answer for me